### PR TITLE
NPM: Fix missed lerna peer dependency update

### PR DESCRIPTION
**Pull request by feelepxyz**
Fixes Sentry: https://sentry.io/organizations/dependabot/issues/432196320/events/46702018625/

Issue was created by a double update on the same package.json for the same dependency and version where the second update didn't need to update anything. 

This happened when a dependency was nested with a peer requirement. Previously we always run the peer dependency checkers on the root package json as well as any package.json's referenced by the dependency requirements. But the yarn peer dependency checker always only runs on the requirement package json meaning that the yarn peer dependency checker runs twice on the same nested package json for both the root package and nested packge (because it's referenced in dep.requirements).

Fixed this by only running the peer dependency checker on package.json's referenced in dep.requirements so the checker only runs once in this case.

This also slims down the files requiring update (used in file updater) to only include the root yarn lockfile if it looks like a yarn workspace lockfile with the dependency in it.

This should also speed up large lerna projects where the root package/lockfiles are huge and don't need updating.

### CODE REVIEW

File: npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker/version_resolver.rb [hunk 1]
```diff
@@ -408,10 +416,6 @@ def write_lock_files
           end
         end
 
-        def prepared_yarn_lockfile_content(content)
-          content.gsub(/^#{Regexp.quote(dependency.name)}\@.*?\n\n/m, "")
```
Comment (feelepxyz) [inline npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker/version_resolver.rb]: seems like a copy from the sub_dependency version resolver

File: npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker/version_resolver.rb [hunk 2]
```diff
@@ -238,12 +237,21 @@ def relevant_unmet_peer_dependencies
...
             old_unmet_peer_dependencies.any? do |old_issue|
               old_issue.slice(:requirement_name, :requiring_dep_name) ==
                 issue.slice(:requirement_name, :requiring_dep_name)
             end
           end
+
+          # If we find duplicate errors it probably means we have double run the
+          # checker on the same files and want to hear about it
+          if issues.length != issues.uniq.length
+            duplicate = issues.detect { |i| issues.count(i) > 1 }
```
Comment (greysteil) [inline npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker/version_resolver.rb]:
```suggestion
            duplicate = issues.find { |i| issues.count(i) > 1 }
```

File: npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker/version_resolver.rb [hunk 3]
```diff
@@ -408,10 +416,6 @@ def write_lock_files
           end
         end
 
-        def prepared_yarn_lockfile_content(content)
```
Comment (greysteil) [inline npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker/version_resolver.rb]: Any idea what the purpose of these removals was before, and why it's OK to get rid of this?

File: npm_and_yarn/lib/dependabot/npm_and_yarn/dependency_files_filterer.rb [hunk 4]
```diff
@@ -14,23 +15,28 @@ def initialize(dependency_files:, updated_dependencies:)
...
-        dependency_files.select do |file|
-          if manifest?(file)
-            package_manifests.include?(file)
-          elsif lockfile?(file)
-            package_manifests.any? do |package_file|
-              File.dirname(package_file.name) == File.dirname(file.name)
+        @files_requiring_update ||=
+          begin
+            dependency_files.select do |file|
+              if manifest?(file)
```
Comment (greysteil) [inline npm_and_yarn/lib/dependabot/npm_and_yarn/dependency_files_filterer.rb]: I think we want an `if file.support_file?` condition here

File: npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker/version_resolver.rb (hunk 3 above)
Comment (feelepxyz) [reply npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker/version_resolver.rb]: yeah pretty sure it was just a copy from the sub version resolver where it's used to update the sub-dependency

File: npm_and_yarn/lib/dependabot/npm_and_yarn/dependency_files_filterer.rb (hunk 4 above)
Comment (feelepxyz) [reply npm_and_yarn/lib/dependabot/npm_and_yarn/dependency_files_filterer.rb]: Oh actually I can probably remove support files as this is now only used to filter manifests and lockfiles for the version resolvers and file updater.

**Pull request merged**